java/memory/memory-core/src/test/java/org/apache/arrow/memory/TestArrowBuf.java

```diff
@@ -138,12 +129,44 @@ public void testSetBytesBigEndian() {
     assertFalse(data.hasArray());
     assertFalse(data.isDirect());
     assertEquals(ByteOrder.BIG_ENDIAN, data.order());
-    try (ArrowBuf buf = allocator.buffer(expected.length)) {
+    try (BufferAllocator allocator = new RootAllocator(128);
+        ArrowBuf buf = allocator.buffer(expected.length)) {
       buf.setBytes(0, data);
       byte[] actual = new byte[expected.length];
       buf.getBytes(0, actual);
       assertArrayEquals(expected, actual);
     }
   }
 
+  @Test
+  public void testEnabledAssertion() {
+    ((Logger) LoggerFactory.getLogger("org.apache.arrow")).setLevel(Level.TRACE);
+    ClassLoader.getSystemClassLoader().setDefaultAssertionStatus(true);
```

Comment:
We probably want to reset this, too

Reply:
Don't we need to reset it to its original value, instead of assuming what it was?

Reply:
Added

Reply:
How does this test even work, actually? The property is already set by the time you get here.

Reply:
Ohhh thank you, you are right, assertions are enabled by default in all the unit test scope, then I remove the property and only validate the message.